music_assistant/controllers/music.py

```diff
@@ -1053,28 +1053,45 @@ async def get_resume_position(self, media_item: Audiobook | PodcastEpisode) -> t
         Returns a boolean with the fully_played status
         and an integer with the resume position in ms.
         """
+        provider_fully_played = False
+        provider_position_ms = 0
+
+        # Try to get position from providers
         for prov_mapping in media_item.provider_mappings:
-            if not (music_prov := self.mass.get_provider(prov_mapping.provider_instance)):
+            if not (provider := self.mass.get_provider(prov_mapping.provider_instance)):
                 continue
-            with suppress(NotImplementedError):
-                return await music_prov.get_resume_position(
-                    prov_mapping.item_id, media_item.media_type
-                )
-        # no provider info found, fallback to library playlog
-        if db_entry := await self.mass.music.database.get_row(
-            DB_TABLE_PLAYLOG,
-            {
-                "media_type": media_item.media_type.value,
-                "item_id": media_item.item_id,
-                "provider": media_item.provider,
-            },
-        ):
-            resume_position_ms = (
-                db_entry["seconds_played"] * 1000 if db_entry["seconds_played"] else 0
+            # Type guard: ensure this is a MusicProvider with get_resume_position method
+            if hasattr(provider, "get_resume_position") and callable(provider.get_resume_position):
```

Comment:
why not simply assert isinstance MusicProvider here ?

Reply:
Fixed in new PR